### DISCUSSION TITLE
v2.2.0

### DISCUSSION THREAD

**Discussion by gianni-rosato** [gianni-rosato/svt-av1-psy#64] [Announcements]
Long time no see! We're back with another exciting release: **SVT-AV1-PSY v2.2.0**! 🎉

### Features
- `--tf-strength` parameter for manually adjusting temporal filtering strength. By default, reduces temporal filtering strength by 4x
- `--chroma-qm-min` & `--chroma-qm-max` parameters for adjusting quantization matrix flatness for chroma planes
- `--tune 4` - A new Tune called Still Picture has been introduced for AVIF encoding, with promising gains observed over aomenc, aomenc 4:4:4, and mainline SVT-AV1

### Quality & Performance
- Enable 10-bit subpixel motion estimation processing for 10-bit inputs on Preset 3 & below
- Enable SUPERRES_AUTO_ALL for Tune 3 when superres-mode=4
  - Disables ALT-REF superres for speed and quality
- Add unipred bias for SSIM full mode decision cost when using Tune 3
- New perceptual QM curve when using Tune 3
  - Better chroma QM adjustment & better high fidelity performance
- Vastly improved screen content detection when `--scm 2` is set. Currently only enabled in Tune 4

### Documentation
- `--help` menu correctly prints maximum CRF value (70)
- Default Tune 2 is better reflected across documentation
- Updated build documentation
- Additional featured projects

### Bug Fixes
- Scoped down QM disablement to only affect NEON-enabled systems
- Scoped down global sharpness features to `--sharpness 1`, as was the original intention

### API updates
- No API changes on this release

### Encoder
- Improve the tradeoffs for the random access mode across presets:
  - Speedup of ~15% across presets M0 - M8 while maintaining similar quality levels
- Improve the tradeoffs for the low-delay mode across presets
- Increased temporal resolution setting to 6L for 4k resolutions by default
- Added ARM optimizations for functions with `c_only` equivalent yielding an average speedup of ~13% for 4k 10-bit

## Cleanup Build & bug fixes & documentation
- Profile-guided-optimized helper build overhaul
- Major cleanup and fixing of Neon unit test suite
- Address stylecheck dependence on public repositories

## Support Us

As SVT-AV1-PSY's codebase has become more complex and the encoder's capabilities have increased dramatically, our efforts have scaled in kind. We have poured hours into coding, testing, distributing, and supporting this piece of open-source software entirely for free, and our work isn't stopping any time soon.

If you appreciate the work that we do and you'd like to support us, we are always excited to see code contributions from outside of the core development team. Otherwise, you can support us monetarily via the links below.

**Gianni Rosato**: [Donate](https://github.com/sponsors/gianni-rosato?o=esc)
**Julio Barba**: *Coming Soon*
**BlueSwordM**: *Coming Soon*
**Clybius**: *Coming Soon*

Any support you can offer goes a long way, and we sincerely appreciate it. ♥️

## Binaries

Linux builds: *(MD5: `Coming Soon`)*

macOS builds: *(MD5: `Coming Soon`)*

Windows builds: *(MD5: `Coming Soon`)*

*Thanks for using SVT-AV1-PSY!* <3

**Full Changelog**: https://github.com/gianni-rosato/svt-av1-psy/commits/v2.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/gianni-rosato/svt-av1-psy/releases/tag/v2.2.0'>v2.2.0</a>.</em>

**Comment by gitoss** [comment]
Are these (esp. the Windows binary) build with PGO - or did you try benchmarking Visual Studio (w/o PGO, if it doesn't work for VS) vs MSYS-based (w/ PGO)?

The log for the respective commit theorizes that gcc is faster than clang, but this probably is very dependent on the system. https://gitlab.com/AOMediaCodec/SVT-AV1/-/merge_requests/2254

Btw, I've managed to understand how to compile SVT-AV1 with Visual Studio and CLANG, but didn't benchmark it yet (latest CLANG 19 can be added manually, VS standard is 17).

**Reply by Uranite** [reply]
The windows binary is built with ClangCL with these parameters `-DENABLE_AVX512=ON -DBUILD_SHARED_LIBS=OFF -DCMAKE_CXX_FLAGS_RELEASE="-flto /O2 /DNDEBUG -march=x86-64-v3" -DCMAKE_C_FLAGS_RELEASE="-flto /O2 /DNDEBUG -march=x86-64-v3"`
MSYS2's Clang is slower than ClangCL/Clang in windows
btw, in that same merge request it was found that clang is actually [faster](https://gitlab.com/AOMediaCodec/SVT-AV1/-/merge_requests/2254#note_2008517387) than gcc and it was [rectified](https://gitlab.com/AOMediaCodec/SVT-AV1/-/commit/ec593019c1c09148d20fd93d0077059fad1a01a9) in further commit

**Reply by gitoss** [reply]
> MSYS2's Clang is slower than ClangCL/Clang in windows btw

True that: From my benchmark (Zen4), MSYS2' clang64 even with PGO enabled is still slower than VS using Clang - even though PGO isn't supported with VS yet. 

The differences between compilers (Microsoft, gcc, llvm) and base systems (MSYS2, VS) are in single digit percentages - which is still significant for video encoding, esp. because SVT-AV1 already uses CPU-specific SIMD code.

**Comment by DesertCookie** [comment]
Is it expected for SVT-AV1-PSY to be slower than "vanilla" SVT-AV1 with the same settings? I've seen a slowdown of just over 1 fps on my 5950X dropping encoding speed to 2 fps from just over 3 fps; that's with preset 4 and RF 20-24, 1x1 tiling on a 4K 10 bit in-/output.

**Reply by gianni-rosato** [reply]
this could depend on so many things
